CBinding/Compiler/ICompiler.cs

```diff
@@ -30,47 +30,431 @@
 // WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 //
 
+using System;
+using System.Threading.Tasks;
+using System.IO;
+using System.Text;
+using System.Linq;
+
 using Mono.Addins;
 
 using MonoDevelop.Core;
 using MonoDevelop.Projects;
+using MonoDevelop.Core.Execution;
+
 
 namespace CBinding
 {
-	[TypeExtensionPoint ("/CBinding/Compilers")]
 	public interface ICompiler
 	{
 		string Name {
 			get;
 		}
-		
+
 		Language Language {
 			get;
 		}
-		
+
 		string CompilerCommand {
 			get;
 		}
-		
+
 		bool SupportsCcache {
 			get;
 		}
-		
+
 		bool SupportsPrecompiledHeaders {
 			get;
 		}
-		
+
 		string GetCompilerFlags (Project project, CProjectConfiguration configuration);
-		
+
 		string GetDefineFlags (Project project, CProjectConfiguration configuration);
-		
+
 		BuildResult Compile (
-		    Project project,
-		    ProjectFileCollection projectFiles,
-		    ProjectPackageCollection packages,
-		    CProjectConfiguration configuration,
-		    ProgressMonitor monitor);
-		
+			Project project,
+			ProjectFileCollection projectFiles,
+			ProjectPackageCollection packages,
+			CProjectConfiguration configuration,
+			ProgressMonitor monitor);
+
 		void Clean (ProjectFileCollection projectFiles, CProjectConfiguration configuration, ProgressMonitor monitor);
 	}
-}
+
+	[TypeExtensionPoint("/CBinding/Toolchains")]
+	public abstract class CMakeToolchain : FolderBasedProject
```

Comment:
Yes, that BaseDirectory can actually be removed and hence inheritance, since only File name should be passed as a parameter to `results.AddWarning()` method , as i understand from its signature , though it works fine for both the cases :)